com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -11,364 +11,217 @@ namespace MLAPI.Prototyping
-                    {
-                        //Snap, set T to 1 (100% of the lerp)
-                        m_LerpTime = 1f;
-                    }
-
-                    float sendDelay = (IsServer || !EnableRange || !AssumeSyncedSends || NetworkManager.ConnectedClients[NetworkManager.LocalClientId].PlayerObject == null) ? (1f / FixedSendsPerSecond) : GetTimeForLerp(transform.position, NetworkManager.ConnectedClients[NetworkManager.LocalClientId].PlayerObject.transform.position);
-                    m_LerpTime += Time.unscaledDeltaTime / sendDelay;
 
-                    if (ExtrapolatePosition && Time.unscaledTime - m_LastReceiveTime < sendDelay * MaxSendsToExtrapolate)
-                    {
-                        transform.position = Vector3.LerpUnclamped(m_LerpStartPos, m_LerpEndPos, m_LerpTime);
-                    }
-                    else
-                    {
-                        transform.position = Vector3.Lerp(m_LerpStartPos, m_LerpEndPos, m_LerpTime);
-                    }
+            SetupVar(m_NetworkPosition, m_Transform.position, ref m_OldPosition);
+            SetupVar(m_NetworkRotation, m_Transform.rotation, ref m_OldRotation);
+            SetupVar(m_NetworkWorldScale, m_Transform.lossyScale, ref m_OldScale);
 
-                    if (ExtrapolatePosition && Time.unscaledTime - m_LastReceiveTime < sendDelay * MaxSendsToExtrapolate)
-                    {
-                        transform.rotation = Quaternion.SlerpUnclamped(m_LerpStartRot, m_LerpEndRot, m_LerpTime);
-                    }
-                    else
-                    {
-                        transform.rotation = Quaternion.Slerp(m_LerpStartRot, m_LerpEndRot, m_LerpTime);
-                    }
-                }
-            }
-
-            if (IsServer && EnableRange && EnableNonProvokedResendChecks)
+            m_PositionChangedDelegate = GetOnValueChangedDelegate<Vector3>(current =>
             {
-                CheckForMissedSends();
-            }
-        }
+                transform.position = current;
+                m_OldPosition = current;
+            });
+            m_NetworkPosition.OnValueChanged += m_PositionChangedDelegate;
+            m_RotationChangedDelegate = GetOnValueChangedDelegate<Quaternion>(current =>
+            {
+                transform.rotation = current;
+                m_OldRotation = current;
+            });
+            m_NetworkRotation.OnValueChanged += m_RotationChangedDelegate;
+            m_ScaleChangedDelegate = GetOnValueChangedDelegate<Vector3>(current =>
+            {
+                SetWorldScale(current);
+                m_OldScale = current;
+            });
+            m_NetworkWorldScale.OnValueChanged += m_ScaleChangedDelegate;
 
-        [ClientRpc]
-        private void ApplyTransformClientRpc(Vector3 position, Vector3 eulerAngles, ClientRpcParams rpcParams = default)
-        {
-            if (enabled)
+            if (IsServer)
```

Comment:
I'm pretty sure write permissions should be set on both the client and the server. Because they do not get automatically updated if you change the server permission and the client might still want to access this.

Reply:
good point. plus I don't really see any reason to keep that check